app/controllers/users_controller.rb

```diff
@@ -7,25 +7,13 @@ def new
 
   def create
     @user = User.new(user_params)
+    result = UserSignUp.new(@user).sign_up(params[:stripeToken], params[:invitation_token])
 
-    if @user.valid?
-      token = params[:stripeToken]
-      charge = StripeWrapper::Charge.create(
-        amount: 999,
-        source: token,
-        description: "Subscription fee for #{@user.email_address}"
-      )
-      if charge.successful?
-        @user.save
-        handle_invitation
-        AppMailer.delay.send_welcome_email(@user)
-        flash[:success] = "Thank you for registering with Myflix"
-        redirect_to signin_path
-      else
-        flash.now[:danger] = charge.error_message
-        render :new
-      end
+    if result.successful?
+      flash[:success] = "Thank you for registering with Myflix. Please sign in now."
+      redirect_to signin_path
     else
+      flash[:danger] = result.error_message
```

Comment:
Make sure you use `flash.now` when you use `render`. `flash` will survive the next page render as well.